packages/ListBox/src/components/WithTags/WithTags.js

```diff
@@ -0,0 +1,157 @@
+  /** Callback once a pill is remove from the Trigger */
+  onRemove: PropTypes.func,
+  /** Regex that match the input of the user and reports to onCustomOption. The default is a basic email regex */
+  regexCustomOption: PropTypes.instanceOf(RegExp),
+  /** Render prop to override the default Pill style, see example for it's uses.  */
+  renderPill: PropTypes.func,
+  /** An array of id that helps the ListBoxWithTags to known what elements are selected  */
+  selectedOptions: PropTypes.arrayOf(PropTypes.shape({})),
+};
+
+const defaultProps = {
+  filter: undefined,
+  noResultsMessage: null,
+  onChange: () => {},
+  onCustomOption: null,
+  onRemove: () => {},
+  regexCustomOption: /^.+@.+\..+$/,
+  renderPill: null,
+  selectedOptions: null,
+};
+
+const renderTrigger = ({ t, size, selectedOptions, onRemove, renderPill }) => (...args) => {
+  const [, , , attributes] = args;
+  const { propsForTrigger, refTrigger, dispatch, types, handleKeyDown, handleKeyUp } = attributes;
+
+  function handleClick(event) {
+    event.stopPropagation();
+    if (event.target.dataset.pkaAnchor === "listbox-tags-pill-delete") {
+      return;
+    }
+
+    if (event.key === "Enter" || event.key === " ") return;
+    dispatch({ type: types.togglePopover });
+  }
+
+  const handleRemove = option => event => {
+    event.stopPropagation();
+    onRemove(option);
+  };
+
+  return (
+    <sc.Trigger
+      ref={refTrigger}
+      {...propsForTrigger()}
+      onClick={handleClick}
+      onKeyUp={handleKeyUp}
+      onKeyDown={handleKeyDown}
+      size={size}
+    >
+      <Pills>
+        {selectedOptions.map(item => {
+          if (typeof renderPill === "function") {
+            return renderPill({ option: item, Pill, onRemove: handleRemove(item) });
+          }
+
+          return (
+            <Pill key={item.label} onRemove={handleRemove(item)} size={size}>
+              {item.label}
+            </Pill>
+          );
+        })}
+        {selectedOptions.length ? null : (
+          <sc.PlaceHolder>
+            <sc.PlaceHolderText>{t("listBoxWithTags.placeholder")}</sc.PlaceHolderText>
+          </sc.PlaceHolder>
+        )}
+      </Pills>
+    </sc.Trigger>
+  );
+};
+
+export default function WithTags(props) {
+  const {
+    children,
+    filter,
+    noResultsMessage,
+    onChange,
+    onCustomOption,
+    onRemove,
+    regexCustomOption,
+    renderPill,
+    selectedOptions,
+    ...moreProps
+  } = props;
+  const { t } = useI18n();
+
+  const refDivRoot = React.useRef(null);
+  /* eslint-disable react/prop-types */
+  const size =
+    typeof props.size !== "undefined" &&
+    [ListBox.types.size.MEDIUM, ListBox.types.size.SMALL, ListBox.types.size.LARGE].includes(props.size)
+      ? props.size
+      : ListBox.types.size.MEDIUM;
+  /* eslint-enable react/prop-types */
+
+  const refFilter = React.useRef(null);
+
+  function handleKeyDown(event) {
+    const label = event.target.value;
+    const regexEmail = regexCustomOption;
```

Comment:
You don't need the variable `regexEmail`, and who says it will always be an `email` regular expression?